### PR TITLE
Fix/push pull windows

### LINKED ISSUE #291: push and pull fail, GIT_TERMINAL_PROMPT=0 command not found
A fresh install of the jupyter-lab git extension, stage and commit on an existing git repository works fine. However, pressing the button `push` fails with a message just saying `Push failed`. In the terminal protocol, it says 
```
[E 13:29:11.800 LabApp] 500 POST /git/push?1544963351615 (::1) 124.80ms referer=http://localhost:8916/lab
```
Pressing button `pull` fails with a message `The command ""GIT_TERMINAL_PROMPT=0 git pull --no-commit"" is either misspelled or could not be found.`. No terminal message protocolled. 

My Git by default uses saved credentials, so there is no need to prompt for username/password.
I never had problems using git on my system, both with the terminal commands or with third-party software, so it must be an issue related to jupyter lab. The command `GIT_TERMINAL_PROMPT=0 git pull --no-commit` works fine when I execute it in my shell (cygwin's mintty). It does not work (CommandNotFound) in PowerShell, however, which opens when I go to Git>Open terminal in the jupyter lab menu. But I never use PowerShell for anything, so I see no reason why PowerShell is used here.

Does jupyter lab/git use the PowerShell? I this the reason for the commands to fail (it looks to me that PowerShell cannot handle prefixes like `GIT_TERMINAL_PROMPT=0` before the actual command... Can I tell jupyter lab/git not to use PowerShell?

I am on Windows 7, Python 3.6.0 and Git 2.11.1 installations (not under cygwin), and jupyter lab 0.35.4.

### PR DESCRIPTION
**Pull request by ameier38**
- fixes #291 by passing in env dictionary
- separates commands when passing to Popen

### CODE REVIEW

**Comment by ameier38** [comment]
@jaipreet-s let me know if you have any comments. Would like to get this merged as I cannot currently use on Windows.

**Comment by jaipreet-s** [comment]
Hi @ameier38 - Thanks for picking this up! I'll get back on this ASAP.

File: tests/unit/test_pushpull.py [hunk 1]
```diff
@@ -93,6 +103,8 @@ def test_git_push_fail(mock_subproc_popen):
 def test_git_push_success(mock_subproc_popen):
     # Given
     process_mock = Mock()
+    env = os.environ.copy()
```
Comment (jaipreet-s) [inline tests/unit/test_pushpull.py]: Only commend I have is that if we could enhance these tests:

We could mock out `os.environ` to return a dummy value and then assert that subprocess was called with the dummy value + `GIT_TERMINAL_PROMPT` as the env parameter

Comment (ameier38) [reply tests/unit/test_pushpull.py]: good catch, updated.

**Comment by jaipreet-s** [comment]
Thanks!

**Pull request merged**